### PR TITLE
feat(build): Build binary for consumers

### PR DESCRIPTION
**Pull request by nikhars**
- Adds makefile target to build all project files and build release targets.
- Changes to allow building binary for consumers. Code in examples
directory cannot be built with `--release` option. We need to build
binaries with that option for performance runs.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,6 +3,13 @@ format:
 	cargo +stable fmt --all
 .PHONY: format
 
+build: setup-git
+	cargo +stable build
+.PHONY: build
+
+release: setup-git
+	@cargo +stable build --release
```
Comment (volokluev) [inline Makefile:11]: do you need the `@` here?

**Comment by volokluev** [comment]
Did you try this on a local kafka?

**Comment by nikhars** [comment]
> Did you try this on a local kafka?

Not yet. In this PR I just wanted to have the ability to create a binary using the rust_arroyo library. This should unblock things like creating Docker image etc. I would be working on modifying the `errors.rs` so that it consumes from the errors topic

File: Makefile (hunk 1 above)
Comment (nikhars) [reply Makefile]: Removed it. I copied this from relay so it was left behind.

**Review by volokluev — APPROVED** [review]

**Pull request merged**